### ISSUE TITLE
1.6 -> 1.7 secret-upgrade failure

### ISSUE THREAD

**Issue by monopole** [kubernetes/kubernetes#47762]
[gubernator](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gke-gci-1-6-cvm-1-7-upgrade-cluster-new/24#secret-upgrade)
[history](https://k8s-testgrid.appspot.com/1.6-1.7-upgrade#gke-gci-cvm-upgrade-cluster-new&include-filter-by-regex=%5EOverall%24%7Csecret%5C-upgrade)

```
Expected error:
    <*errors.errorString | 0xc420d8d750>: {
        s: "expected \"mode of file \\\"/etc/secret-volume/data\\\": -rw-r--r--\" in container output: Expected\n    <string>: mode of file \"/etc/configmap-volume/data\": -rw-r--r--\n    content of file \"/etc/secret-volume/data\": keep it secret\n    \nto contain substring\n    <string>: mode of file \"/etc/secret-volume/data\": -rw-r--r--",
    }
    expected "mode of file \"/etc/secret-volume/data\": -rw-r--r--" in container output: Expected
        <string>: mode of file "/etc/configmap-volume/data": -rw-r--r--
        content of file "/etc/secret-volume/data": keep it secret
        
    to contain substring
        <string>: mode of file "/etc/secret-volume/data": -rw-r--r--
not to have occurred

k8s.io/kubernetes/test/e2e/framework.(*Framework).testContainerOutputMatcher(0xc42084e460, 0x37b931a, 0x16, 0xc420dca000, 0x0, 0xc420ee2e88, 0x2, 0x2, 0x38b3010)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2224 +0x240
k8s.io/kubernetes/test/e2e/framework.(*Framework).TestContainerOutput(0xc42084e460, 0x37b931a, 0x16, 0xc420dca000, 0x0, 0xc420ee2e88, 0x2, 0x2)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:400 +0x8a
k8s.io/kubernetes/test/e2e/upgrades.(*SecretUpgradeTest).testPod(0x5685088, 0xc42084e460)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/upgrades/secrets.go:142 +0x782
k8s.io/kubernetes/test/e2e/upgrades.(*SecretUpgradeTest).Test(0x5685088, 0xc42084e460, 0xc4213a8420, 0x2)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/upgrades/secrets.go:69 +0x90
k8s.io/kubernetes/test/e2e.(*chaosMonkeyAdapter).Test(0xc421652700, 0xc421158e60)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/cluster_upgrade.go:205 +0x1ed
k8s.io/kubernetes/test/e2e.(*chaosMonkeyAdapter).Test-fm(0xc421158e60)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/cluster_upgrade.go:270 +0x34
k8s.io/kubernetes/test/e2e/chaosmonkey.(*chaosmonkey).Do.func1(0xc421158e60, 0xc420480640)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/chaosmonkey/chaosmonkey.go:89 +0x76
created by k8s.io/kubernetes/test/e2e/chaosmonkey.(*chaosmonkey).Do
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/chaosmonkey/chaosmonkey.go:90 +0x1c7from junit_upgradeupgrades.xml
```

/sig storage
/assign @msau42 

Please redirect as needed, thanks Michelle!

**Comment by msau42** [comment]
This is a strange flaky test.  For some reason it got the output of the config map pod instead of the secret pod.

**Comment by monopole** [comment]
@msau42  We've cut 1.7 - is there anyone who can say if this flake a matter of concern?

It both passed and failed on the 21st, and is now stale

**Comment by msau42** [comment]
I tried to triage this failure, and I am very stumped at this point.  It seems like either the arguments or output of two different tests running in parallel got mixed up.  The output is displaying one line that is from the configmap test, and another line that is from the secrets test.  I have not been able to narrow down which layer could be the culprit: golang, test framework, or the kubernetes system.

The test only flaked this way once so far.  So I would put this at low priority unless we see it more often.

**Comment by msau42** [comment]
It's still been flaking periodically :(  I'll try to run this test repeatedly to try to reproduce.

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gke-gci-1-6-cvm-1-7-upgrade-cluster-new/273

**Comment by msau42** [comment]
The test doesn't seem to be flaking in this way anymore.

/close
